src/jira/jira.service.ts

```diff
@@ -16,9 +16,18 @@ export class JiraService {
 
   // Default API connection for Jira is the same as for Confluence
   constructor(private http: HttpService, private config: ConfigService) {
-    this.baseUrl = this.config.get('confluence.baseURL');
-    this.apiUsername = this.config.get('confluence.apiUsername');
-    this.apiToken = this.config.get('confluence.apiToken');
+    this.init();
+  }
+
+  private init(component = '') {
+    if (component !== '' && component === 'jiraIssues') {
```

Comment:
it's not dynamic function, You shoudn't use conditional in that way, let's imagine someone another will create jiraIssues component name and he has not credentials etc